worlds/messenger/__init__.py

```diff
@@ -41,38 +42,42 @@ class MessengerWorld(World):
         "Crest": {"Sun Crest", "Moon Crest"},
         "Phobe": set(PHOBEKINS),
         "Phobekin": set(PHOBEKINS),
-        "Shuriken": {"Windmill Shuriken"},
     }
 
     option_definitions = messenger_options
 
     base_offset = 0xADD_000
     item_name_to_id = {item: item_id
                        for item_id, item in enumerate(ALL_ITEMS, base_offset)}
-    mega_shard_locs = [shard for region in MEGA_SHARDS for shard in MEGA_SHARDS[region]]
+    seal_locs = [seal for seals in SEALS.values() for seal in seals]
+    mega_shard_locs = [shard for shards in MEGA_SHARDS.values() for shard in shards]
+    shop_locs = [f"The Shop - {shop_loc}" for shop_loc in SHOP_ITEMS]
     location_name_to_id = {location: location_id
                            for location_id, location in
                            enumerate([
                                *ALWAYS_LOCATIONS,
-                               *SEALS,
+                               *seal_locs,
                                *mega_shard_locs,
                                *BOSS_LOCATIONS,
+                               *shop_locs,
+                               *FIGURINES,
+                               "Money Wrench",
                            ], base_offset)}
 
-    data_version = 2
-    required_client_version = (0, 3, 9)
+    data_version = 3
+    required_client_version = (0, 4, 0)
 
     web = MessengerWeb()
 
     total_seals: int = 0
     required_seals: int = 0
-    
-    @classmethod
-    def stage_assert_generate(cls, multiworld: MultiWorld) -> None:
-        for player in multiworld.get_game_players(cls.game):
-            player_name = multiworld.player_name[player] = multiworld.get_player_name(player).replace("_", " ")
-            if not all(c.isalnum() or c in "- " for c in player_name):
-                raise ValueError(f"Player name {player_name} is not alpha-numeric.")
```

Comment:
Since you remove this, how did you solve entering "odd" names?

Reply:
a config file gets created in the game folder now and it's mentioned in both the setup guide and the info page

Reply:
good thing i read docs :D